src/fhirpath.js

```diff
@@ -505,7 +523,13 @@ engine.FunctionInvocation = function(ctx, parentData, node) {
   const fnName = args[0];
   args.shift();
   var rawParams = args && args[0] && args[0].children;
-  return doInvoke(ctx, fnName, parentData, rawParams);
+  if (fnName == 'defineVariable') {
+    return doInvoke(ctx, fnName, parentData, rawParams);
+  }
+  let ctxForArgs = { ...ctx };
```

Comment:
I think that a variable defined in a subexpression for one function parameter should not affect another function parameter.
Here's a new unit test to test this:
```js
  it('defineVariable in a function parameters', () => {
    let expr = "defineVariable(defineVariable('param','ppp').select(%param), defineVariable('param','value').select(%param)).select(%ppp)";
    let f = fhirpath.compile(expr, r4_model);
    expect(f(input.patientExample))
      .toStrictEqual(["value"]);
  });
```

Reply:
These tests have been added and are now passing too!